### PR TITLE
Update javascript.md

### PR DESCRIPTION
**Pull request by PiusNyakoojo**
This fixes a potential typo.

In example [19.3](https://github.com/VML/styleguide/blob/master/javascript.md#whitespace--around-keywords), a space after the function name is a rule often referred to as ```"space-before-function-paren"```. This rule is more of a stylistic preference; whereas, I believe, the intention for rule [19.3](https://github.com/VML/styleguide/blob/master/javascript.md#whitespace--around-keywords) is to guide users to not leave a space before the function parentheses of a *function call*.

### CODE REVIEW

**Comment by andrewicarlson** [comment]
The intention here is to stylistically guide away from a space between a function name and it's argument list in [both the call and the declaration](https://github.com/VML/styleguide/blob/master/javascript.md#whitespace--around-keywords):

> Place no space between the argument list and the function name in function calls **and declarations**

The adjustment made here changes the "bad" example to be identical to the "good" example.

**Comment by PiusNyakoojo** [comment]
Ahh, you're right. I didn't catch the "**and declarations**" part.

However, the example for ```good``` and ```bad``` practices remains distinct. The difference lies with the space following the ```log``` function name.

```javascript
// bad
function fight() {
  console.log ('Swoosh!');
}

// good
function fight() {
  console.log('Swoosh!');
}
```

[7.11](https://github.com/VML/styleguide/blob/master/javascript.md#functions--signature-spacing) recommends the rule ```space-before-function-paren``` which supports (and perhaps [encourages](http://eslint.org/docs/rules/space-before-function-paren)) a space after the function name declaration.

Applying this rule, the following code is ```good``` practice. But of course it's important to remain consistent. The same code is ```bad``` practice when its environment follows a different convention.

```javascript
function fight () {
  console.log('Swoosh!');
}
```

In conclusion, there seems to be a conflict between [7.11](https://github.com/VML/styleguide/blob/master/javascript.md#functions--signature-spacing) and [19.3](https://github.com/VML/styleguide/blob/master/javascript.md#whitespace--around-keywords).

* 7.11 *accepts* a space after the function name declaration.
* 19.3 *rejects* a space after the function name declaration.

The change introduced here suggests a removal of the "**and declarations**" part of 19.3. The presented conflict will be resolved, as a result.

**Comment by andrewicarlson** [comment]
> 7.11 accepts a space after the function name declaration.
> 19.3 rejects a space after the function name declaration.

7.11 accepts a space after the anonymous function *signature* but not the name.

```
// good
const x = function () {}; // This is an anonymous function signature, space accepted.
const y = function a() {}; // This is a function name, no space accepted between name and argument list.
```

19.3 says:

> Place no space between the argument list and the function name

Which would line up with 19.3's 'bad' example:

```
// bad
function fight () { // Bad because there is a space between the name and argument list.
  console.log ('Swooosh!');
}
```
